### PR TITLE
[somfytahoma] Improved handling of target temperature command

### PR DESCRIPTION
**Pull request by lolodomo**
Signed-off-by: Laurent Garnier <lg.hc@free.fr>

### CODE REVIEW

**Comment by lolodomo** [comment]
@octa22 ; please review, this is a fix for better handling of target temperature command. It now accepts a value with any temperature unit or a value without any unit.

File: bundles/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/internal/handler/SomfyTahomaBaseThingHandler.java [hunk 1]
```diff
@@ -387,12 +393,7 @@ public void updateThingStatus(List<SomfyTahomaState> states) {
     }
 
     private @Nullable SomfyTahomaState getStatusState(List<SomfyTahomaState> states) {
-        for (SomfyTahomaState state : states) {
-            if (STATUS_STATE.equals(state.getName()) && state.getType() == TYPE_STRING) {
-                return state;
-            }
-        }
-        return null;
+        return SomfyTahomaDevice.getState(states, STATUS_STATE, TYPE_STRING);
```
Comment (octa22) [inline bundles/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/internal/handler/SomfyTahomaBaseThingHandler.java:396]: Please see below. I do not want to move the binding logic to model classes.

File: bundles/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/internal/model/SomfyTahomaDevice.java [hunk 2]
```diff
@@ -67,4 +68,18 @@ public SomfyTahomaDeviceDefinition getDefinition() {
...
+
+    public static @Nullable SomfyTahomaState getState(List<SomfyTahomaState> states, String stateName,
+            @Nullable Integer stateType) {
+        for (SomfyTahomaState state : states) {
+            if (stateName.equals(state.getName()) && (stateType == null || stateType == state.getType())) {
+                return state;
+            }
+        }
+        return null;
+    }
```
Comment (octa22) [inline bundles/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/internal/model/SomfyTahomaDevice.java:84]: This is a model class. Please keep is as a simple POJO

File: bundles/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/internal/handler/SomfyTahomaBaseThingHandler.java (hunk 1 above)
Comment (lolodomo) [reply bundles/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/internal/handler/SomfyTahomaBaseThingHandler.java:396]: Ok, I moved the new method `getState` into `SomfyTahomaBaseThingHandler`.

File: bundles/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/internal/model/SomfyTahomaDevice.java (hunk 2 above)
Comment (lolodomo) [reply bundles/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/internal/model/SomfyTahomaDevice.java:84]: Ok, I moved the method into `SomfyTahomaBaseThingHandler`.

**Review by octa22 — APPROVED** [review]
LGTM

**Comment by lolodomo** [comment]
@fwolter : please review

**Review by fwolter — APPROVED** [review]

**Pull request merged**